src/sentry/api/endpoints/project_group_index.py

```diff
@@ -855,21 +855,26 @@ def put(self, request, project):
             group_list_by_times_seen = sorted(group_list, key=lambda x: x.times_seen, reverse=True)
             primary_group, groups_to_merge = group_list_by_times_seen[0], group_list_by_times_seen[1:]
 
+            group_ids_to_merge = sorted([g.id for g in groups_to_merge])
```

Comment:
I forgot we sort by `times_seen` already (right above). I just wanted something deterministic to test against, I can probably remove this. Though I'm almost tempted to reverse `groups_to_merge` so small groups are merged/deleted first...